dot/import_test.go

```diff
@@ -2,3 +2,200 @@
+	require.NoError(t, err)
+
+	stateFP := setupStateFile(t)
+	headerFP := setupHeaderFile(t)
+
+	type args struct {
+		basepath  string
+		stateFP   string
+		headerFP  string
+		firstSlot uint64
+	}
+	tests := []struct {
+		name string
+		args args
+		err  error
+	}{
+		{
+			name: "no arguments",
+			err:  errors.New("read .: is a directory"),
+		},
+		{
+			name: "working example",
+			args: args{
+				basepath:  basepath,
+				stateFP:   stateFP,
+				headerFP:  headerFP,
+				firstSlot: 262493679,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := ImportState(tt.args.basepath, tt.args.stateFP, tt.args.headerFP, tt.args.firstSlot)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func Test_newHeaderFromFile(t *testing.T) {
+	t.Parallel()
+
+	type args struct {
+		filename string
+	}
+	tests := []struct {
+		name string
+		args args
+		want *types.Header
+		err  error
+	}{
+		{
+			name: "working example",
+			args: args{filename: setupHeaderFile(t)},
+			want: &types.Header{
+				ParentHash:     common.MustHexToHash("0x3b45c9c22dcece75a30acc9c2968cb311e6b0557350f83b430f47559db786975"),
+				Number:         big.NewInt(1482002),
+				StateRoot:      common.MustHexToHash("0x09f9ca28df0560c2291aa16b56e15e07d1e1927088f51356d522722aa90ca7cb"),
+				ExtrinsicsRoot: common.MustHexToHash("0xda26dc8c1455f8f81cae12e4fc59e23ce961b2c837f6d3f664283af906d344e0"),
+			},
+		},
+		{
+			name: "no arguments",
+			err:  errors.New("read .: is a directory"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := newHeaderFromFile(tt.args.filename)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+
+			if tt.want != nil {
+				assert.Equal(t, tt.want.ParentHash, got.ParentHash)
+				assert.Equal(t, tt.want.Number, got.Number)
+				assert.Equal(t, tt.want.StateRoot, got.StateRoot)
+				assert.Equal(t, tt.want.ExtrinsicsRoot, got.ExtrinsicsRoot)
+			}
+		})
+	}
+}
+
+func Test_newTrieFromPairs(t *testing.T) {
+	t.Parallel()
+
+	type args struct {
+		filename string
+	}
+	tests := []struct {
+		name string
+		args args
+		want common.Hash
+		err  error
+	}{
+		{
+			name: "no arguments",
+			err:  errors.New("read .: is a directory"),
+			want: common.Hash{},
+		},
+		{
+			name: "working example",
+			args: args{filename: setupStateFile(t)},
+			want: common.MustHexToHash("0x09f9ca28df0560c2291aa16b56e15e07d1e1927088f51356d522722aa90ca7cb"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := newTrieFromPairs(tt.args.filename)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+
+			if !tt.want.IsEmpty() {
+				assert.Equal(t, tt.want, got.MustHash())
+			}
+		})
+	}
+}
+
+func setupStateFile(t *testing.T) string {
```

Comment:
is this specific to kusama? A comment about what type of data this is would be nice.